rfd/0019-cluster-routing.md

```diff
@@ -0,0 +1,44 @@
+---
+authors: Russell Jones (rjones@goteleport.com)
+state: draft
+---
+
+# RFD 19 - Cluster Routing
+
+## What
+
+This RFD proposes changes to how Teleport issues certificates and routes SSH requests within clusters. These changes would make using certificate issued by a root cluster directly on leaf clusters substantially easier.
+
+## Why
+
+At the moment connections to a leaf cluster typically go through a root cluster. If the latency to the root cluster is high, accessing servers within a leaf cluster leads to a frustrating user experience. Allowing direct use of certificates issued by a root cluster on a leaf cluster mitigates this problem.
+
+Teleport supports this use case.
+
+Users can run `tsh login clusterName` to obtain a certificate with `RouteToCluster` set to a leaf cluster then use `tsh ssh -J clusterName serverName` to bypass the root cluster. However, switching between multiple leaf clusters is cumbersome as `tsh login clusterName` has to be issued each time the user wishes to switch between clusters. It also makes working on multiple clusters concurrently almost impossible.
+
+## Original Solution: jumphost address to cluster name
```

Comment:
nit: put the latest design first and add a `## Alternative / discarded designs` section at the bottom.
when someone new wants to read the design in the future, no need to walk them through the whole history, just the final outcome